src/charm.py

```diff
@@ -198,6 +211,21 @@ def _get_hydra_endpoint_info(self) -> Optional[str]:
 
         return oauth2_provider_url
 
+    def _get_login_ui_endpoint_info(self, key: str) -> Optional[str]:
+        if self.model.relations[self._login_ui_relation_name]:
```

Comment:
The login_ui_endpoints library handles this, no need to check it here.

Reply:
done. (included LoginUIEndpointsRelationMissingError handling)